File: GremsReliquary/src/main/java/gremsReliquary/relics/cursed/UnbalancedScales.java
package gremsReliquary.relics.cursed;

import com.badlogic.gdx.graphics.Texture;
import com.evacipated.cardcrawl.mod.stslib.relics.ClickableRelic;
import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.actions.common.RelicAboveCreatureAction;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.helpers.PowerTip;
import com.megacrit.cardcrawl.powers.DexterityPower;
import com.megacrit.cardcrawl.powers.StrengthPower;
import com.megacrit.cardcrawl.rooms.AbstractRoom;
import gremsReliquary.GremsReliquary;
import gremsReliquary.relics.AbstractGremRelic;
import gremsReliquary.util.TextureLoader;

public class UnbalancedScales extends AbstractGremRelic implements ClickableRelic {
    public static final String ID = GremsReliquary.makeID(UnbalancedScales.class.getSimpleName());
    public static final Texture IMG = TextureLoader.getTexture("gremsReliquaryResources/images/relics/UnbalancedScales.png");
    public static final Texture OUTLINE = TextureLoader.getTexture("gremsReliquaryResources/images/relics/outline/UnbalancedScales.png");
    public static int amount = 2;
    
    AbstractCreature p = AbstractDungeon.player;
    private static boolean usedThisCombat = false;
    
    public UnbalancedScales() {
        super(ID, IMG, OUTLINE, RelicTier.UNCOMMON, RelicType.CURSED, LandingSound.CLINK);
        tips.clear();
        tips.add(new PowerTip(name, description));
    }
    
    @Override
    public void atBattleStart() {
        beginLongPulse();
        usedThisCombat = false;
        AbstractDungeon.actionManager.addToTop(new RelicAboveCreatureAction(AbstractDungeon.player, this));
        flash();
        act(new ApplyPowerAction(p, p, new StrengthPower(p, 5), 5));
        act(new ApplyPowerAction(p, p, new DexterityPower(p, -5), -5));
    }
    
    // Description
    @Override
    public String getUpdatedDescription() {
        return DESCRIPTIONS[0];
    }
    
    @Override
    public void onVictory() {
        stopPulse();
    }
    
    @Override
    public void onRightClick() {
        if (!usedThisCombat && AbstractDungeon.getCurrRoom().phase == AbstractRoom.RoomPhase.COMBAT) {
            AbstractDungeon.actionManager.addToTop(new RelicAboveCreatureAction(AbstractDungeon.player, this));
            stopPulse();
            act(new ApplyPowerAction(p, p, new StrengthPower(p, -10), -10));
            act(new ApplyPowerAction(p, p, new DexterityPower(p, 10), 10));
            usedThisCombat = true;
        }
    }
}

File: GremsReliquary/src/main/kotlin/kotlinReliquary/relics/cursed/UnbalancedScales.kt
package kotlinReliquary.relics.cursed

import com.evacipated.cardcrawl.mod.stslib.relics.ClickableRelic
import com.megacrit.cardcrawl.actions.common.ApplyPowerAction
import com.megacrit.cardcrawl.actions.common.RelicAboveCreatureAction
import com.megacrit.cardcrawl.core.AbstractCreature
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.powers.DexterityPower
import com.megacrit.cardcrawl.powers.StrengthPower
import com.megacrit.cardcrawl.rooms.AbstractRoom
import gremsReliquary.GremsReliquary
import gremsReliquary.util.TextureLoader
import kotlinReliquary.relics.AbstractGremRelic


class UnbalancedScales : AbstractGremRelic(ID, IMG, OUTLINE, RelicTier.UNCOMMON, AbstractGremRelic.RelicType.CURSED, LandingSound.CLINK), ClickableRelic {
     var p: AbstractCreature? = AbstractDungeon.player

    companion object {
        val ID = GremsReliquary.makeID(UnbalancedScales::class.java.simpleName)
        val IMG = TextureLoader.getTexture("gremsReliquaryResources/images/relics/UnbalancedScales.png")
        val OUTLINE = TextureLoader.getTexture("gremsReliquaryResources/images/relics/outline/UnbalancedScales.png")
        var amount = 2
        private var usedThisCombat = false
    }

    init {
        p = AbstractDungeon.player
    }

    override fun atBattleStart() {
        beginLongPulse()
        usedThisCombat = false
        AbstractDungeon.actionManager.addToTop(RelicAboveCreatureAction(AbstractDungeon.player, this))
        flash()
        act(ApplyPowerAction(p, p, StrengthPower(p, 5), 5))
        act(ApplyPowerAction(p, p, DexterityPower(p, -5), -5))
    }

    override fun onRightClick() {
        if (!usedThisCombat && AbstractDungeon.getCurrRoom().phase == AbstractRoom.RoomPhase.COMBAT) {
            AbstractDungeon.actionManager.addToTop(RelicAboveCreatureAction(AbstractDungeon.player, this))
            stopPulse()
            act(ApplyPowerAction(p, p, StrengthPower(p, -10), -10))
            act(ApplyPowerAction(p, p, DexterityPower(p, 10), 10))
            usedThisCombat = true
        }
    }

    override fun onVictory() {
        stopPulse()
    }

    override fun getUpdatedDescription(): String {
        return DESCRIPTIONS[0]
    }
}